classes/StoredPasswordsDAO.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+/**
+ * @file plugins/generic/betterPassword/classes/StoredPasswordsDAO.php
+ *
+ * @class StoredPasswordsDAO
+ * @brief Database operations with the StoredPasswords
+ */
+namespace APP\plugins\generic\betterPassword\classes;
+
+use PKP\core\EntityDAO;
+use APP\plugins\generic\betterPassword\classes\StoredPasswords as StoredPasswords;
+use PKP\plugins\Hook;
+use Illuminate\Support\Facades\DB;
+
+class StoredPasswordsDAO extends EntityDAO {
+	/** @copydoc EntityDAO::$schema */
+	public $schema = 'storedPassword';
+
+	/** @copydoc EntityDAO::$schemaName */
+	public $schemaName = 'storedPassword';
+
+	/** @copydoc EntityDAO::$table */
+	public $table = 'stored_passwords';
+	
+	/** @copydoc EntityDAO::$primaryKeyColumn */
+	public $primaryKeyColumn = 'id';
+	
+	/** @copydoc EntityDAO::$primaryTableColumn */
+	public $primaryTableColumns = [
+		'id' => 'id',
+		'user_id' => 'user_id',
+		'password' => 'password',
+		'lastChangeTime' => 'last_change_time'
+	];
+
+	/**
+	 * Constructor
+	 */
+	public function __construct()
+	{
+		parent::__construct(new \PKP\services\PKPSchemaService());
+	}
+
+	/**
+	 * Overrides EntityDao's newDataObject()
+	 * @return StoredPasswords New StoredPasswords object
+	 */
+	public function newDataObject () {
+		return new StoredPasswords();
+	}
+
+	/**
+	 * Create a new Stored Passwords object
+	 * @param int $user_id The user's user id
+	 * @param string $password The user's password list
+	 * @param \DateTime $last_change_time Time of the last password change
+	 * @return StoredPasswords New StoredPasswords object
+	 */
+	public function newDataObjects (int $user_id, string $password = '', \DateTime $last_change_time = new \DateTime ('now')) : ?StoredPasswords {
```

Comment:
Is there any suggestions for an accurate alternate name? I can't just name it newDataObject because of the one override function and newDataObjectWithDefaults seems too long.

Reply:
It would probably be better to forgo this method entirely.  In the three instances you are using it, the only thing it is consistently doing is forcing the last changed time to the current datetime.

The callers could just do:
```
$storedPasswords = $storedPasswordsDao->newDataObject();
$storedPasswords->setUserId($user->getId());
$storedPasswords->setChangeTime(now());
```

and 

```
$storedPasswords = $storedPasswordsDao->newDataObject();
$storedPasswords->setUserId($user->getId());
$storedPasswords->setPasswords($password, true);
```